### PR TITLE
enhance: Decouple compaction from shard

### PR DESCRIPTION
**Pull request by bigsheeper**
Decouple compaction from shard, remove dependencies on shards (e.g. SyncSegments, injection).

issue: https://github.com/milvus-io/milvus/issues/32809

### CODE REVIEW

File: internal/datacoord/compaction_trigger.go [hunk 1]
```diff
@@ -530,18 +521,11 @@ func (t *compactionTrigger) handleSignal(signal *compactionSignal) {
 			break
 		}
 		start := time.Now()
-		if err := fillOriginPlan(t.allocator, plan); err != nil {
+		if err := fillOriginPlan(collectionID, t.handler, t.allocator, plan); err != nil {
```
[the remark this answers is not in the document]
Comment (XuanYang-cn) [reply internal/datacoord/compaction_trigger.go:524]: See L:492

File: internal/datanode/compactor.go [hunk 2]
```diff
@@ -429,7 +410,18 @@ func (t *compactionTask) compact() (*datapb.CompactionPlanResult, error) {
 	ctx, span := otel.Tracer(typeutil.DataNodeRole).Start(t.ctx, fmt.Sprintf("Compact-%d", t.getPlanID()))
 	defer span.End()
 
-	log := log.Ctx(ctx).With(zap.Int64("planID", t.plan.GetPlanID()), zap.Int32("timeout in seconds", t.plan.GetTimeoutInSeconds()))
+	if len(t.plan.GetSegmentBinlogs()) < 1 {
+		log.Warn("compact wrong, there's no segments in segment binlogs")
```
Comment (XuanYang-cn) [inline internal/datanode/compactor.go:414]: this warn should attach PlanID

File: internal/datacoord/compaction_trigger.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bigsheeper) [reply internal/datacoord/compaction_trigger.go]: done

File: internal/datanode/compactor.go (hunk 2 above)
Comment (bigsheeper) [reply internal/datanode/compactor.go]: updated

File: internal/datanode/compaction/mix_compactor_test.go [hunk 3]
```diff
@@ -204,39 +178,32 @@ func (s *MixCompactionTaskSuite) TestCompactDupPK() {
...
-			MinPK:    s.segWriter.pkstats.MinPk,
-			MaxPK:    s.segWriter.pkstats.MaxPk,
-		}
-		bfs := metacache.NewBloomFilterSet(statistic)
+		//statistic := &storage.PkStatistics{
+		//	PkFilter: s.segWriter.pkstats.BF,
+		//	MinPK:    s.segWriter.pkstats.MinPk,
+		//	MaxPK:    s.segWriter.pkstats.MaxPk,
+		//}
+		//bfs := metacache.NewBloomFilterSet(statistic)
```
Comment (XuanYang-cn) [inline internal/datanode/compaction/mix_compactor_test.go:193]: Meybe remove those comment codes

Comment (bigsheeper) [reply internal/datanode/compaction/mix_compactor_test.go:193]: I will make it in the next pr

**Pull request merged**